app/models/team.rb

```diff
@@ -40,4 +40,12 @@ def add_member(user_id)
   def remove_member(user_id)
     team_members.find_by(user_id: user_id)&.destroy
   end
+
+  def messages
+    account.messages.where(conversation_id: conversations.pluck(:id))
```

Comment:
To improve query: we can have index on `index_messages_on_account_id_and_conversation_id unique: true`
I ran explain on the above query on my local and might be expensive in the future with a lot of records:
```Sort  (cost=45.05..46.00 rows=381 width=245)
   Sort Key: created_at
   ->  Seq Scan on messages  (cost=0.00..28.72 rows=381 width=245)
         Filter: ((account_id = 1) AND (conversation_id = ANY ('{33,16,18,1,9,2,19,32,8,10,11,5,6,17,4,13,34,30,15,12,3,7,14}'::integer[])))
(4 rows)```

We can take it later in the tech debt also just have it mentioned somewhere we should unique: true index on the combination of the conversation_id and account_id in the message table.

Reply:
Let's take the improvements as separate tech debt.

Reply:
as a part of the tech debt we can also revisit the specs around reports and see how we can improve the overall execution time there.